File: spec/php/RepeatEosU4Test.php
<?php
namespace Kaitai\Struct\Tests;

class RepeatEosU4Test extends TestCase {
    public function testRepeatEosU4() {
        $r = RepeatEosU4::fromFile(self::SRC_DIR_PATH . "/repeat_eos_struct.bin");
        $this->assertEquals([0, 0x42, 0x42, 0x815], $r->numbers());
    }
}

File: spec/go/repeat_eos_u4_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestRepeatEosU4(t *testing.T) {
	f, err := os.Open("../../src/repeat_eos_struct.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r RepeatEosU4
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, []uint{0, 0x42, 0x42, 0x815}, r.Numbers)
}
